electron/src/main.ts

```diff
@@ -283,6 +300,56 @@ const handleAppEvents = () => {
     isQuitting = true;
   });
 
+  app.on('login', async (event, webContents, request, authInfo, callback) => {
+    if (authInfo.isProxy) {
+      if (session.defaultSession) {
+        session.defaultSession.allowNTLMCredentialsForDomains(authInfo.host);
+      }
+
+      event.preventDefault();
+
+      if (authenticatedProxyInfo) {
+        const {username, password} = authenticatedProxyInfo;
+        logger.info('Sending provided credentials to authenticated proxy ...');
+        return callback(username, password);
+      }
+
+      const systemProxy = await getProxySettings();
+      const systemProxySettings = systemProxy && (systemProxy.http || systemProxy.https);
+
+      if (systemProxySettings) {
+        const {
+          credentials: {username, password},
+        } = systemProxySettings;
+        authenticatedProxyInfo = new URL(
+          `http${systemProxy.https ? 's' : ''}://${username}:${password}@${authInfo.host}`,
+        );
+        return callback(username, password);
+      }
+
+      ipcMain.once(
+        EVENT_TYPE.PROXY_PROMPT.SUBMITTED,
+        (event: IpcMessageEvent, promptData: {password: string; username: string}) => {
+          const {username, password} = promptData;
+
+          logger.log('Proxy prompt was submitted');
+          authenticatedProxyInfo = new URL(`http://${username}:${password}@${authInfo.host}`);
```

Comment:
So we are not supporting https proxies for now?

Reply:
We are now :)